I have run the system test and I have the error message below.  I suspect the file you are referring to does not exist in my bucket. Can you create it in the `setUp` method?
```
[2020-04-22 02:43:49,073] {taskinstance.py:1186} ERROR - {'code': 9, 'message': 'Execution failed: generic::failed_precondition: while running "gsutil cp gs://test-airflow-life-sciences/input.in /tmp": unexpected exit status 1 was not ignored'}
Traceback (most recent call last):
  File "/opt/airflow/airflow/models/taskinstance.py", line 1024, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/airflow/providers/google/cloud/operators/life_sciences.py", line 74, in execute
    project_id=self.project_id)
  File "/opt/airflow/airflow/providers/google/common/hooks/base_google.py", line 345, in inner_wrapper
    return func(self, *args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/hooks/life_sciences.py", line 101, in run_pipeline
    self._wait_for_operation_to_complete(operation_name)
  File "/opt/airflow/airflow/providers/google/cloud/hooks/life_sciences.py", line 148, in _wait_for_operation_to_complete
    raise AirflowException(str(error))
airflow.exceptions.AirflowException: {'code': 9, 'message': 'Execution failed: generic::failed_precondition: while running "gsutil cp gs://test-airflow-life-sciences/input.in /tmp": unexpected exit status 1 was not ignored'}
```

--- a/airflow/providers/google/cloud/example_dags/example_life_sciences.py
+++ b/airflow/providers/google/cloud/example_dags/example_life_sciences.py
@@ -1,0 +1,72 @@
+import os
+
+from airflow import models
+from airflow.utils import dates
+from airflow.providers.google.cloud.operators.life_sciences import LifeSciencesRunPipelineOperator
+
+PROJECT_ID = os.environ.get("GCP_PROJECT_ID", "example-id")
+BUCKET = os.environ.get("GCP_GCS_BUCKET", "test-life-sciences-bucket")
+LOCATION = os.environ.get("GCP_LOCATION", 'example-project-location')
+
+SIMPLE_ACTION_PIEPELINE = {
+        "pipeline": {
+            "actions": [
+                {
+                    "imageUri": "bash",
+                    "commands": ["-c", "echo Hello, world"]
+                },
+            ],
+            "resources": {
+                "regions": ["us-central11"],
+                "virtualMachine": {
+                    "machineType": "n1-standard-1",
+                }
+            }
+        }
+}
+MULTI_ACTION_PIPELINE = {
+        "pipeline": {
+            "actions": [
+                {
+                    "imageUri": "google/cloud-sdk",
+                    "commands": ["gsutil", "cp", "gs://{}/input.in".format(BUCKET), "/tmp"]
+                },
+                {
+                    "imageUri": "bash",
+                    "commands": ["-c", "sha1sum /tmp/in > /tmp/test.sha1"]
+                },
+                {
+                    "imageUri": "google/cloud-sdk",
+                    "commands": ["gsutil", "cp", "/tmp/output.sha1", "gs://{}/output.sha1".format(BUCKET)]
+                },
+            ],
+            "resources": {
+                "regions": ["{}".format(LOCATION)],
+                "virtualMachine": {
+                    "machineType": "n1-standard-1",
+                }
+            }
+        }
+}
+
+with models.DAG("example_gcp_life_sciences",
+                default_args=dict(start_date=dates.days_ago(1)),
+                schedule_interval=None,
+                tags=['example'],
+                ) as dag:
+
+    simple_life_science_action_pipeline = LifeSciencesRunPipelineOperator(
+        task_id='simple-action-pipeline',
+        body=SIMPLE_ACTION_PIEPELINE,
+        project_id=PROJECT_ID,
+        location=LOCATION
+    )
+
+    multiple_life_science_action_pipeline = LifeSciencesRunPipelineOperator(
+        task_id='multi-action-pipeline',
+        body=MULTI_ACTION_PIPELINE,
+        project_id=PROJECT_ID,
+        location=LOCATION
+    )
+
+    simple_life_science_action_pipeline >> multiple_life_science_action_pipeline

--- a/airflow/providers/google/cloud/hooks/life_sciences.py
+++ b/airflow/providers/google/cloud/hooks/life_sciences.py
@@ -1,0 +1,147 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Hook for Google Cloud Life Sciences service"""
+
+import time
+from typing import Any, Dict, Optional
+
+import google.api_core.path_template
+from googleapiclient.discovery import build
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+
+# Time to sleep between active checks of the operation results
+TIME_TO_SLEEP_IN_SECONDS = 10
+
+
+# noinspection PyAbstractClass
+class LifeSciencesHook(GoogleBaseHook):
+    """
+    Hook for the Google Cloud Life Sciences APIs.
+
+    All the methods in the hook where project_id is used must be called with
+    keyword arguments rather than positional.
+
+    :param api_version: API version used (for example v1 or v1beta1).
+    :type api_version: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    _conn = None  # type: Optional[Any]
+
+    def __init__(
+        self,
+        api_version: str = "v2beta",
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None
+    ) -> None:
+        super().__init__(gcp_conn_id, delegate_to)
+        self.api_version = api_version
+
+    def get_conn(self):
+        """
+        Retrieves the connection to Cloud Life Sciences.
+
+        :return: Google Cloud Life Sciences service object.
+        """
+        if not self._conn:
+            http_authorized = self._authorize()
+            self._conn = build("lifesciences", self.api_version,
+                               http=http_authorized, cache_discovery=False)
+        return self._conn
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def run_pipeline(self, body: Dict, location: str, project_id: str):
+        """
+        Runs a pipeline
+
+        :param body: The request body.
+        :type body: dict
+        :param location: The location of the project. For example: "us-east1".
+        :type location: str
+        :param project_id: Optional, Google Cloud Project project_id where the function belongs.
+            If set to None or missing, the default project_id from the GCP connection is used.
+        :type project_id: str
+        :return: Dict
+        """
+        parent = self._location_path(project_id=project_id, location=location)
+        service = self.get_conn()
+
+        request = service.projects().locations().\
+            pipelines().run(parent=parent, body=body)
+
+        response = request.execute(num_retries=self.num_retries)
+
+        # wait
+        operation_name = response['name']
+        self._wait_for_operation_to_complete(operation_name)
+
+        return response
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def _location_path(self, project_id: str, location: str):
+        """
+        Return a location string.
+
+        :param project_id: :param project_id: Optional, Google Cloud Project project_id where the function belongs.
+            If set to None or missing, the default project_id from the GCP connection is used.
+        :type project_id: str
+        :param location: The location of the project. For example: "us-east1".
+        :type location: str
+        """
+        return google.api_core.path_template.expand(
+            'projects/{project}/locations/{location}',
+            project=project_id,
+            location=location,
+        )
+
+    def _wait_for_operation_to_complete(self, operation_name: str) -> None:
+        """
+        Waits for the named operation to complete - checks status of the
+        asynchronous call.
+
+        :param operation_name: The name of the operation.
+        :type operation_name: str
+        :return: The response returned by the operation.
+        :rtype: dict
+        :exception: AirflowException in case error is returned.
+        """
+        service = self.get_conn()
+        while True:
+            operation_response = service.projects().locations().\
+                operations().get(name=operation_name).\
+                execute(num_retries=self.num_retries)
+
+            if operation_response.get("done"):
+                response = operation_response.get("response")
+                error = operation_response.get("error")
+                # Note, according to documentation always either response or error is
+                # set when "done" == True
+                if error:
+                    raise AirflowException(str(error))
+                return response
+            time.sleep(TIME_TO_SLEEP_IN_SECONDS)
+
+
+

--- a/airflow/providers/google/cloud/hooks/life_sciences.py
+++ b/airflow/providers/google/cloud/hooks/life_sciences.py
@@ -27,7 +27,7 @@ from airflow.exceptions import AirflowException
 from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
 
 # Time to sleep between active checks of the operation results
-TIME_TO_SLEEP_IN_SECONDS = 10
+TIME_TO_SLEEP_IN_SECONDS = 5
 
 
 # noinspection PyAbstractClass
@@ -132,7 +132,7 @@ class LifeSciencesHook(GoogleBaseHook):
             operation_response = service.projects().locations().\
                 operations().get(name=operation_name).\
                 execute(num_retries=self.num_retries)
-
+            self.log.info('Waiting for pipeline operation to complete')
             if operation_response.get("done"):
                 response = operation_response.get("response")
                 error = operation_response.get("error")

--- a/airflow/providers/google/cloud/operators/life_sciences.py
+++ b/airflow/providers/google/cloud/operators/life_sciences.py
@@ -1,0 +1,71 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Operators that interact with Google Cloud Life Sciences service."""
+
+from typing import Iterable, Optional
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.google.cloud.hooks.life_sciences import LifeSciencesHook
+from airflow.utils.decorators import apply_defaults
+
+
+class LifeSciencesRunPipelineOperator(BaseOperator):
+    """
+    Runs a pipeline
+
+    :param body: The request body
+    :param location: The location of the project
+    :param project_id: ID of the Google Cloud project if None then
+        default project_id is used.
+    The connection ID to use to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param api_version: API version used (for example v2beta).
+    :type api_version: str
+    """
+
+    template_fields = ("body", "gcp_conn_id", "api_version")  # type: Iterable[str]
+
+    @apply_defaults
+    def __init__(self,
+                 body: dict,
+                 location: str,
+                 project_id: Optional[str] = None,
+                 gcp_conn_id: str = "google_cloud_default",
+                 api_version: str = "v2beta",
+                 *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.body = body
+        self.location = location
+        self.project_id = project_id
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self._validate_inputs()
+
+    def _validate_inputs(self):
+        if not self.body:
+            raise AirflowException("The required parameter 'body' is missing")
+        if not self.location:
+            raise AirflowException("The required parameter 'location' is missing")
+
+    def execute(self, context):
+        hook = LifeSciencesHook(gcp_conn_id=self.gcp_conn_id, api_version=self.api_version)
+
+        return hook.run_pipeline(body=self.body,
+                                 location=self.location,
+                                 project_id=self.project_id)

--- a/docs/howto/operator/gcp/life_sciences.rst
+++ b/docs/howto/operator/gcp/life_sciences.rst
@@ -1,0 +1,82 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+
+Google Cloud Life Sciences Operators
+====================================
+The `Google Cloud Life Sciences <https://cloud.google.com/life-sciences/>`__ is a service that executes
+series of compute engine containers on the google cloud. It is used to process, analyze and annotate genomics
+and biomedical data at scale.
+
+.. contents::
+  :depth: 1
+  :local:
+
+
+Prerequisite Tasks
+^^^^^^^^^^^^^^^^^^
+
+.. include:: _partials/prerequisite_tasks.rst
+
+
+Pipeline Configuration
+^^^^^^^^^^^^^^^^^^^^^^
+In order to run the pipeline, it is necessary to configure the request body.
+Here is an example of the pipeline configuration with a single action.
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_life_sciences.py
+    :language: python
+    :dedent: 0
+    :start-after: [START howto_configure_simple_action_pipeline]
+    :end-before: [END howto_configure_simple_action_pipeline]
+
+The pipeline can also be configured with multiple action.
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_life_sciences.py
+    :language: python
+    :dedent: 12
+    :start-after: [START howto_configure_multiple_action_pipeline]
+    :end-before: [END howto_configure_multiple_action_pipeline]
+
+Read about the `request body parameters <https://cloud.google.com/life-sciences/docs/reference/rest/v2beta/projects.locations.pipelines/run?authuser=1#request-body/>`__
+to understand all the fields you can include in the configuration
+
+.. _howto/operator:LifeSciencesRunPipelineOperator:
+
+LifeSciencesRunPipelineOperator
+-------------------------------
+Use the
+:class:`~airflow.providers.google.cloud.operators.life_sciences.LifeSciencesRunPipelineOperator`
+to execute pipelines.
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_life_sciences.py
+    :language: python
+    :dedent: 12
+    :start-after: [START howto_run_pipeline]
+    :end-before: [END howto_run_pipeline]
+
+Reference
+^^^^^^^^^
+
+For further information, look at:
+
+* `Client Library Documentation <https://cloud.google.com/life-sciences/docs/reference/rest>`__
+* `Product Documentation <https://cloud.google.com/life-sciences/docs/>`__
+
+
+

--- a/docs/operators-and-hooks-ref.rst
+++ b/docs/operators-and-hooks-ref.rst
@@ -659,6 +659,10 @@ These integrations allow you to perform various operations within the Google Clo
      - :mod:`airflow.providers.google.cloud.hooks.kms`
      -
      -
+   * - `Cloud Life Sciences <https://cloud.google.com/life-sciences/>`__
+     - :doc:`How to use <howto/operator/gcp/life_sciences>`
+     - :mod:`airflow.providers.google.cloud.hooks.life_sciences`
+     - :mod:`airflow.providers.google.cloud.operators.life_sciences`
 
    * - `Kubernetes Engine <https://cloud.google.com/kubernetes_engine/>`__
      -

--- a/tests/providers/google/cloud/hooks/test_life_sciences.py
+++ b/tests/providers/google/cloud/hooks/test_life_sciences.py
@@ -1,0 +1,302 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Tests for Google Cloud Life Sciences Hook
+"""
+import unittest
+from typing import Optional
+from unittest import mock
+
+from mock import PropertyMock
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.cloud.hooks.life_sciences import LifeSciencesHook
+from tests.providers.google.cloud.utils.base_gcp_mock import (
+    GCP_PROJECT_ID_HOOK_UNIT_TEST, mock_base_gcp_hook_default_project_id,
+    mock_base_gcp_hook_no_default_project_id,
+)
+
+TEST_OPERATION = {"name": 'operation-name', "metadata": {"@type": 'anytype'},
+                  "done": True, "response": "response"}
+
+TEST_WAITING_OPERATION = {"done": False, "response": "response"}
+TEST_DONE_OPERATION = {"done": True, "response": "response"}
+TEST_ERROR_OPERATION = {"done": True, "response": "response", "error": "error"}
+TEST_PROJECT_ID = "life-science-project-id"
+TEST_LOCATION = 'test-location'
+
+
+class TestLifeSciencesHookWithPassedProjectId(unittest.TestCase):
+
+    def setUp(self):
+        with mock.patch(
+            "airflow.providers.google.common.hooks.base_google.GoogleBaseHook.__init__",
+            new=mock_base_gcp_hook_default_project_id,
+        ):
+            self.hook = LifeSciencesHook(gcp_conn_id="test")
+
+    def test_location_path(self):
+        path = 'projects/life-science-project-id/locations/test-location'
+        path2 = self.hook._location_path(project_id=TEST_PROJECT_ID,
+                                         location=TEST_LOCATION)
+        self.assertEqual(path, path2)
+
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook._authorize")
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.build")
+    def test_life_science_client_creation(self, mock_build, mock_authorize):
+        result = self.hook.get_conn()
+        mock_build.assert_called_once_with(
+            'lifesciences', 'v2beta', http=mock_authorize.return_value, cache_discovery=False
+        )
+        self.assertEqual(mock_build.return_value, result)
+        self.assertEqual(self.hook._conn, result)
+
+    @mock.patch(
+        'airflow.providers.google.common.hooks.base_google.GoogleBaseHook.project_id',
+        new_callable=PropertyMock,
+        return_value=GCP_PROJECT_ID_HOOK_UNIT_TEST
+    )
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook.get_conn")
+    def test_run_pipeline_immediately_complete(self, get_conn_mock, mock_project_id):
+        service_mock = get_conn_mock.return_value
+
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .pipelines.return_value \
+            .run.return_value \
+            .execute.return_value = TEST_OPERATION
+
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .operations.return_value \
+            .get.return_value \
+            .execute.return_value = TEST_DONE_OPERATION
+
+        result = self.hook.run_pipeline(body={},
+                                        location=TEST_LOCATION,
+                                        project_id=TEST_PROJECT_ID)
+        parent = self.hook._location_path(project_id=TEST_PROJECT_ID,
+                                          location=TEST_LOCATION)
+        service_mock.projects.return_value.locations.return_value\
+            .pipelines.return_value.run \
+            .assert_called_once_with(body={},
+                                     parent=parent)
+        self.assertEqual(result, TEST_OPERATION)
+
+    @mock.patch(
+        'airflow.providers.google.common.hooks.base_google.GoogleBaseHook.project_id',
+        new_callable=PropertyMock,
+        return_value=GCP_PROJECT_ID_HOOK_UNIT_TEST
+    )
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook.get_conn")
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.time.sleep")
+    def test_waiting_operation(self, _, get_conn_mock, mock_project_id):
+        service_mock = get_conn_mock.return_value
+
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .pipelines.return_value \
+            .run.return_value \
+            .execute.return_value = TEST_OPERATION
+
+        execute_mock = mock.Mock(
+            **{"side_effect": [TEST_WAITING_OPERATION, TEST_DONE_OPERATION]}
+        )
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .operations.return_value \
+            .get.return_value \
+            .execute = execute_mock
+
+        result = self.hook.run_pipeline(body={},
+                                        location=TEST_LOCATION,
+                                        project_id=TEST_PROJECT_ID)
+        self.assertEqual(result, TEST_OPERATION)
+
+    @mock.patch(
+        'airflow.providers.google.common.hooks.base_google.GoogleBaseHook.project_id',
+        new_callable=PropertyMock,
+        return_value=GCP_PROJECT_ID_HOOK_UNIT_TEST
+    )
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook.get_conn")
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.time.sleep")
+    def test_error_operation(self, _, get_conn_mock, mock_project_id):
+        service_mock = get_conn_mock.return_value
+
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .pipelines.return_value \
+            .run.return_value \
+            .execute.return_value = TEST_OPERATION
+
+        execute_mock = mock.Mock(**{"side_effect": [TEST_WAITING_OPERATION, TEST_ERROR_OPERATION]})
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .operations.return_value \
+            .get.return_value \
+            .execute = execute_mock
+
+        with self.assertRaisesRegex(AirflowException, "error"):
+            self.hook.run_pipeline(body={},
+                                   location=TEST_LOCATION,
+                                   project_id=TEST_PROJECT_ID
+                                   )
+
+
+class TestLifeSciencesHookWithDefaultProjectIdFromConnection(unittest.TestCase):
+
+    def setUp(self):
+        with mock.patch(
+            "airflow.providers.google.common.hooks.base_google.GoogleBaseHook.__init__",
+            new=mock_base_gcp_hook_default_project_id,
+        ):
+            self.hook = LifeSciencesHook(gcp_conn_id="test")
+
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook._authorize")
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.build")
+    def test_life_science_client_creation(self, mock_build, mock_authorize):
+        result = self.hook.get_conn()
+        mock_build.assert_called_once_with(
+            'lifesciences', 'v2beta', http=mock_authorize.return_value, cache_discovery=False
+        )
+        self.assertEqual(mock_build.return_value, result)
+        self.assertEqual(self.hook._conn, result)
+
+    @mock.patch(
+        'airflow.providers.google.common.hooks.base_google.GoogleBaseHook.project_id',
+        new_callable=PropertyMock,
+        return_value=GCP_PROJECT_ID_HOOK_UNIT_TEST
+    )
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook.get_conn")
+    def test_run_pipeline_immediately_complete(self, get_conn_mock, mock_project_id):
+        service_mock = get_conn_mock.return_value
+
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .pipelines.return_value \
+            .run.return_value \
+            .execute.return_value = TEST_OPERATION
+
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .operations.return_value \
+            .get.return_value \
+            .execute.return_value = TEST_DONE_OPERATION
+
+        result = self.hook.run_pipeline(body={},
+                                        location=TEST_LOCATION)
+        parent = self.hook._location_path(location=TEST_LOCATION)
+        service_mock.projects.return_value.locations.return_value\
+            .pipelines.return_value.run \
+            .assert_called_once_with(body={},
+                                     parent=parent)
+        self.assertEqual(result, TEST_OPERATION)
+
+    @mock.patch(
+        'airflow.providers.google.common.hooks.base_google.GoogleBaseHook.project_id',
+        new_callable=PropertyMock,
+        return_value=GCP_PROJECT_ID_HOOK_UNIT_TEST
+    )
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook.get_conn")
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.time.sleep")
+    def test_waiting_operation(self, _, get_conn_mock, mock_project_id):
+        service_mock = get_conn_mock.return_value
+
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .pipelines.return_value \
+            .run.return_value \
+            .execute.return_value = TEST_OPERATION
+
+        execute_mock = mock.Mock(
+            **{"side_effect": [TEST_WAITING_OPERATION, TEST_DONE_OPERATION]}
+        )
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .operations.return_value \
+            .get.return_value \
+            .execute = execute_mock
+
+        result = self.hook.run_pipeline(body={},
+                                        location=TEST_LOCATION)
+        self.assertEqual(result, TEST_OPERATION)
+
+    @mock.patch(
+        'airflow.providers.google.common.hooks.base_google.GoogleBaseHook.project_id',
+        new_callable=PropertyMock,
+        return_value=GCP_PROJECT_ID_HOOK_UNIT_TEST
+    )
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook.get_conn")
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.time.sleep")
+    def test_error_operation(self, _, get_conn_mock, mock_project_id):
+        service_mock = get_conn_mock.return_value
+
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .pipelines.return_value \
+            .run.return_value \
+            .execute.return_value = TEST_OPERATION
+
+        execute_mock = mock.Mock(**{"side_effect": [TEST_WAITING_OPERATION, TEST_ERROR_OPERATION]})
+        service_mock.projects.return_value \
+            .locations.return_value \
+            .operations.return_value \
+            .get.return_value \
+            .execute = execute_mock
+
+        with self.assertRaisesRegex(AirflowException, "error"):
+            self.hook.run_pipeline(body={},
+                                   location=TEST_LOCATION
+                                   )
+
+
+class TestLifeSciencesHookWithoutProjectId(unittest.TestCase):
+
+    def setUp(self):
+        with mock.patch(
+            "airflow.providers.google.common.hooks.base_google.GoogleBaseHook.__init__",
+            new=mock_base_gcp_hook_no_default_project_id,
+        ):
+            self.hook = LifeSciencesHook(gcp_conn_id="test")
+
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook._authorize")
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.build")
+    def test_life_science_client_creation(self, mock_build, mock_authorize):
+        result = self.hook.get_conn()
+        mock_build.assert_called_once_with(
+            'lifesciences', 'v2beta', http=mock_authorize.return_value, cache_discovery=False
+        )
+        self.assertEqual(mock_build.return_value, result)
+        self.assertEqual(self.hook._conn, result)
+
+    @mock.patch(
+        'airflow.providers.google.common.hooks.base_google.GoogleBaseHook.project_id',
+        new_callable=PropertyMock,
+        return_value=None
+    )
+    @mock.patch("airflow.providers.google.cloud.hooks.life_sciences.LifeSciencesHook.get_conn")
+    def test_run_pipeline(self, get_conn_mock, mock_project_id):
+        with self.assertRaises(AirflowException) as e:
+            self.hook.run_pipeline(body={},
+                                   location=TEST_LOCATION)  # pylint: disable=no-value-for-parameter
+
+        self.assertEqual(
+            "The project id must be passed either as keyword project_id parameter or as project_id extra in "
+            "GCP connection definition. Both are not set!",
+            str(e.exception),
+        )

--- a/tests/providers/google/cloud/operators/test_life_sciences.py
+++ b/tests/providers/google/cloud/operators/test_life_sciences.py
@@ -1,0 +1,65 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Tests for Google Life Sciences Run Pipeline operator """
+
+import mock
+import unittest
+
+from airflow.providers.google.cloud.operators.life_sciences import LifeSciencesRunPipelineOperator
+TEST_BODY = {
+    "pipeline": {
+                    "actions": [{}],
+                    "resources": {},
+                    "environment": {},
+                    "timeout": '3.5s'
+    },
+    "labels": "my-pipeline"
+    }
+TEST_OPERATION = {"name": 'operation-name', "metadata": {"@type": 'anytype'},
+                  "done": True, "response": "response"}
+TEST_PROJECT_ID = "life-science-project-id"
+TEST_LOCATION = 'test-location'
+
+
+class TestLifeSciencesRunPipelineOperator(unittest.TestCase):
+
+    @mock.patch("airflow.providers.google.cloud.operators.life_sciences.LifeSciencesHook")
+    def test_executes(self, mock_hook):
+        mock_instance = mock_hook.return_value
+        mock_instance.run_pipeline.return_value = TEST_OPERATION
+        operator = LifeSciencesRunPipelineOperator(
+            task_id='task-id',
+            body=TEST_BODY,
+            location=TEST_LOCATION,
+            project_id=TEST_PROJECT_ID
+        )
+        result = operator.execute(None)
+        self.assertEqual(result, TEST_OPERATION)
+
+    @mock.patch("airflow.providers.google.cloud.operators.life_sciences.LifeSciencesHook")
+    def test_executes_without_project_id(self, mock_hook):
+        mock_instance = mock_hook.return_value
+        mock_instance.run_pipeline.return_value = TEST_OPERATION
+        operator = LifeSciencesRunPipelineOperator(
+            task_id='task-id',
+            body=TEST_BODY,
+            location=TEST_LOCATION,
+        )
+        result = operator.execute(None)
+        self.assertEqual(result, TEST_OPERATION)
+

